### PR TITLE
New resource: `azuread_app_role_assignment`

### PR DESCRIPTION
**Pull request by manicminer**
Depends on: https://github.com/manicminer/hamilton/pull/107

* **New Resource:** `azuread_app_role_assignment`

Closes: #164 
Closes: #230 
Closes: #459

### CODE REVIEW

**Comment by manicminer** [comment]
**Test results**

File: internal/services/approleassignments/app_role_assignment_resource.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+func appRoleAssignmentResourceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client).AppRoleAssignments.AppRoleAssignedToClient
+	servicePrincipalsClient := meta.(*clients.Client).AppRoleAssignments.ServicePrincipalsClient
+
+	appRoleId := d.Get("app_role_id").(string)
+	principalId := d.Get("principal_object_id").(string)
+	resourceId := d.Get("resource_object_id").(string)
+	_, status, err := servicePrincipalsClient.Get(ctx, resourceId, odata.Query{})
+	if err != nil {
```
Comment (katbyte) [inline internal/services/approleassignments/app_role_assignment_resource.go:94]: I THINK THESE LINES CAN BE MERGED?

**Review by katbyte — APPROVED** [review]
aside from one comment LGTM 🏗️

File: internal/services/approleassignments/app_role_assignment_resource.go (hunk 1 above)
Comment (manicminer) [reply internal/services/approleassignments/app_role_assignment_resource.go:94]: nodding-sign

We should get a linter for these :D

**Pull request merged**